components/remote_settings/src/storage.rs

```diff
@@ -51,6 +51,7 @@ impl Storage {
         CREATE TABLE IF NOT EXISTS collection_metadata (
             collection_url TEXT PRIMARY KEY,
             last_modified INTEGER,
+            metadata BLOB NOT NULL,
```

Comment:
I'd rather storing each field metadata separately rather than serializing it into a single field.  I don't feel very strongly about this one, though, feel free to push back on it.

Reply:
I also hesitated there. 

The reason behind my choice is that eventually we will potentially look at other metadata fields, like `attachment` to know whether there's a bundle to pull, or `flags` to know whether a collection is required to have records during startup. 
But, we can perfectly decide to put this aside, and add the additional columns only when we implement these features.

It depends a bit how confident we are in database migration for the future

Reply:
That would definitely help yes please 🙏  Maybe a candidate would be the `fetched` column

Reply:
I got a bit distracted last week, but it's finally up: https://github.com/mozilla/application-services/pull/6537.